app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java

```diff
@@ -512,28 +518,40 @@ private static int getLabelIcon(Attachment.Type type) {
 
     protected void setMediaLabel(List<Attachment> attachments, boolean sensitive,
                                  final StatusActionListener listener) {
-        if (attachments.size() == 0) {
-            mediaLabel.setVisibility(View.GONE);
-            return;
-        }
-        mediaLabel.setVisibility(View.VISIBLE);
-
-        // Set the label's text.
         Context context = itemView.getContext();
-        String labelText = getLabelTypeText(context, attachments.get(0).getType());
-        if (sensitive) {
-            String sensitiveText = context.getString(R.string.status_sensitive_media_title);
-            labelText += String.format(" (%s)", sensitiveText);
-        }
-        mediaLabel.setText(labelText);
+        for (int i = 0; i < mediaLabels.length; i++) {
+            TextView mediaLabel = mediaLabels[i];
+            if (i < attachments.size()) {
+                Attachment attachment = attachments.get(i);
+                mediaLabel.setVisibility(View.VISIBLE);
+                String labelText;
+                String description = attachment.getDescription();
+                if (TextUtils.isEmpty(description)) {
+                    description = context.getString(
+                            R.string.description_status_media_no_description_placeholder);
+                }
+                if (sensitive) {
+                    String sensitiveText = context.getString(R.string.status_sensitive_media_title);
+                    labelText = String.format("%s (%s)", description, sensitiveText);
+                } else {
+                    labelText = description;
+                }
+                mediaLabel.setText(labelText);
 
-        // Set the icon next to the label.
-        int drawableId = getLabelIcon(attachments.get(0).getType());
-        Drawable drawable = AppCompatResources.getDrawable(context, drawableId);
-        ThemeUtils.setDrawableTint(context, drawable, android.R.attr.textColorTertiary);
-        mediaLabel.setCompoundDrawablesWithIntrinsicBounds(drawable, null, null, null);
 
-        mediaLabel.setOnClickListener(v -> listener.onViewMedia(getAdapterPosition(), 0, null));
+                // Set the icon next to the label.
+                int drawableId = getLabelIcon(attachments.get(0).getType());
+                Drawable drawable = AppCompatResources.getDrawable(context, drawableId);
```

Comment:
a simple `context.getDrawable` is enough since our upgrade to minSdk 21 (just tested on a Android 5.0 device)

Reply:
Do we still need `setDrawableTint()` then?